testing out item replacement

briaguya0 opened the pull request:
resolves https://github.com/HarbourMasters/Shipwright/issues/37

briaguya-ai commented:
i [generated a seed](https://ootrandomizer.com/seed/get?id=1104114) with settings that seem like they'll take the least code to get working. seems like a good starting point.

Commit:
skip learning song of storms

briaguya0 committed:
Merge pull request #248 from briaguya-ai/fast-song-of-storms-check

skip learning song of storms

Commit:
don't set flag when getting goron tunic as child

briaguya0 committed:
Merge pull request #249 from briaguya-ai/goron-link-fix

don't set flag when getting goron tunic as child

leggettc18 committed:
Initiates prelude check when master sword unloads.

Not quite how N64 rando does it but so far it's the only way I've found to make it trigger without also triggering the time travel again.

leggettc18 committed:
Merge branch 'testing-out-item-replacement' of https://github.com/briaguya-ai/Shipwright into rando_prelude

leggettc18 committed:
Stops Shadow Temple lore prompts from appearing in rando.

leggettc18 committed:
Skips cutscene of royal tomb explosion in rando.

Explosion sound doesn't play correctly and I think the debris appears in the wrong place, but the functionality is here.

leggettc18 committed:
Improves visual of exploding gravestone.

leggettc18 committed:
Adds some comments explaining the rando differences

aMannus committed:
Skip ruto text box in jabu blue warp

For rando

Commit:
skip intro cutscene in dodongo's cavern

briaguya0 committed:
Merge pull request #253 from briaguya-ai/dc-intro-cs-skip

skip intro cutscene in dodongo's cavern

briaguya0 committed:
Merge pull request #252 from aMannus/jabu-blue-warp-text-skip

Skip ruto text box in jabu blue warp

briaguya0 committed:
Merge pull request #251 from leggettc18/rando-skip-royal-tomb

Rando skip royal tomb

briaguya0 committed:
Merge pull request #247 from aMannus/jabu-ruto-skip

Added flags for ruto falling down hole

Commit:
load spoiler files on boot, fix spoilerfile existing check when making new saves

briaguya0 committed:
Merge pull request #254 from briaguya-ai/better-file-select

file select fixes

Commit:
name entry dropped spoiler logic

briaguya0 committed:
Merge pull request #255 from briaguya-ai/better-file-select

name entry dropped spoiler logic

Commit:
make sure to actually init the cvar

briaguya0 committed:
Merge pull request #256 from briaguya-ai/fix-disable-rando-menu-issue

make sure to actually init the cvar

Commit:
no chime on load

Commit:
uncomment

briaguya0 committed:
Merge pull request #257 from briaguya-ai/no-chime-on-load

no chime on load

aMannus committed:
Skip ganondrof cutscene

Skip to scream part of the death animation, skipping the text boxes etc. For rando

aMannus committed:
Update z_boss_ganondrof.c

briaguya0 committed:
Merge pull request #258 from aMannus/ganondrof-cutscene-skip

Forest Temple Ganondrof cutscene skip

Commit:
skip owl flight cutscenes in rando

briaguya0 committed:
Merge pull request #259 from briaguya-ai/fast-owl

skip owl flight cutscenes in rando

leggettc18 committed:
Fixes skipped text so it only applies to shadow temple.

Earlier fix inadvertently applied to some other text as well, changed logic so that only specified sceneNums and textIds can have this enabled, and text skipped by sceneNum can have the skip overriden by textId if needed. Currently there are no overrides so the textId section of the logic is commented out to avoid compilation errors.

leggettc18 committed:
Adds a default to the switch case statements that leaves the randoSkipText variable unchanged, just in case.

briaguya0 committed:
Merge pull request #250 from leggettc18/rando-shadow-lore-prompt-skip

Stops Shadow Temple lore prompts from appearing in rando.

PurpleHato committed:
TEST: Text for item

Dog committed:
Adding ganon flavor text

leggettc18 committed:
Merge branch 'testing-out-item-replacement' of https://github.com/briaguya-ai/Shipwright into rando_prelude

briaguya0 committed:
Merge pull request #260 from Dog/testing-out-item-replacement

Adding ganon flavor text

briaguya0 committed:
Merge pull request #261 from leggettc18/rando_prelude

Rando prelude v1

PurpleHato committed:
ADD: AMMO Count

Commit:
format ganon text/hint text

Dog committed:
Autoskip the tower cutscene if settings call for tower collapse.

Commit:
ganon hint text logic

briaguya0 committed:
Merge pull request #263 from briaguya-ai/ganon-text-formatting

Ganon text formatting

briaguya0 committed:
Merge pull request #262 from Dog/testing-out-item-replacement

Autoskip the tower cutscene if settings call for tower collapse.

leggettc18 committed:
Improved prelude after time travel fix

Dog committed:
swapped the sizes between ganon hint text and ganon text, as they were set to the wrong things.

briaguya0 committed:
Merge pull request #265 from Dog/testing-out-item-replacement

Fix the ganon buffer sizes

Commit:
this is all i did

Commit:
not the cleanest code ever but it's working

PurpleHato committed:
ADD: GS Count

PurpleHato committed:
Merge branch 'ItemTracker' into this-is-all-i-did

PurpleHato committed:
Merge pull request #13 from briaguya-ai/this-is-all-i-did

this is all i did

PurpleHato committed:
ADD: Wallter (crash for now)

PurpleHato committed:
TWEAK: Wallet check

PurpleHato committed:
FIX: Use DrawItem instread of DrawUpgrade... b-baka!

leggettc18 committed:
Fixes some vanilla bugs introduced by rando code.

aMannus committed:
Added cutscene skip for zelda escaping

Using the debug cutscene skipping function. Also added a conditional so the bridge doesn't spawn closed when cutscene is ready to trigger

PurpleHato committed:
ADD: X Spacing + Placeholders for song

PurpleHato committed:
ADD: default case for items

PurpleHato committed:
TWEAK: Spacing

briaguya0 committed:
Merge pull request #264 from leggettc18/rando_prelude

Improved prelude after time travel fix

PurpleHato committed:
FIX: Light Arrow

PurpleHato committed:
ADD: Ammo Option

Commit:
use groups instead

PurpleHato committed:
Merge pull request #14 from briaguya-ai/groupy-group

use groups instead

PurpleHato committed:
ADD: More spacing logic

Commit:
songs and names

PurpleHato committed:
Merge pull request #15 from briaguya-ai/songs-and-names

songs and names

PurpleHato committed:
TWEAK: Color on wallet

Commit:
colors

PurpleHato committed:
Merge pull request #16 from briaguya-ai/background-color

colors

aMannus committed:
Added flags cutscene before nabooru fight

PurpleHato committed:
ADD: ChromaKey text

briaguya0 committed:
Merge pull request #268 from PurpleHato/ItemTracker

Item tracker V1

aMannus committed:
First attempt skip cs after nabooru defeat

aMannus committed:
Better implementation for specific rando cutscene skips

aMannus committed:
Merge branch 'pr/416' into zelda-escape-cutscene-skip

Commit:
use pulseaudio defaults

Commit:
spaces/tabs

briaguya0 committed:
Merge pull request #269 from briaguya-ai/fix-pulseaudio

Fix pulseaudio

Commit:
move color push/pop to stop crash

Commit:
make the colors work again

briaguya0 committed:
Merge pull request #270 from briaguya-ai/fix-colorstack-crash

Fix colorstack crash

Commit:
the real bottle fix

briaguya0 committed:
Merge pull request #271 from briaguya-ai/the-real-bottle-fix

the real bottle fix

Commit:
pulseaudio values tuned for n64 audio at 44.1khz

Commit:
update tlength

Commit:
remove one hardcoded samplerate

briaguya0 committed:
Merge pull request #272 from briaguya-ai/fix-pulseaudio

Fix pulseaudio

aMannus committed:
Cleaned up and fixed zelda escape skip

The if statement is a freaking monster, but unless we want to skip more cutscenes in the same way later, this is the most compact way of doing it that I know of.

aMannus committed:
Revert one line to match original

nothing functional

aMannus committed:
Merge branch 'pr/416' into spirit-temple-cs-skips-testing

Commit:
another hint line that breaks autonewline logic

briaguya0 committed:
Merge pull request #273 from briaguya-ai/fix-hint-text-that-breaks-geneation

another hint line that breaks autonewline logic

Commit:
don't autospawn epona if we don't have the song/ocarina

briaguya0 committed:
Merge pull request #274 from briaguya-ai/fix-epona-spawn-checks

don't autospawn epona if we don't have the song/ocarina

aMannus committed:
Trying to use iron knuckle death effects

not working yet

aMannus committed:
Streamlined OoT cutscene skip for future additions

Also cleaned up if statement in general

aMannus committed:
Made if statement more readable

Also added clarity for what cutscene was skipped

aMannus committed:
Fixed typo in comment

briaguya0 committed:
Merge pull request #267 from aMannus/zelda-escape-cutscene-skip

Added cutscene skip for zelda escaping

Commit:
Merge branch 'develop' into toir-devmerge

briaguya0 committed:
Merge pull request #275 from briaguya-ai/toir-devmerge

Toir devmerge

aMannus committed:
Janky nabooru defeat cs skip

aMannus committed:
Merge pull request #1 from briaguya-ai/testing-out-item-replacement

Testing out item replacement

aMannus committed:
Merge branch 'rando-base' into spirit-temple-cs-skips-testing

Commit:
altar text formatting (gonna need help shortening some of the french ones)

Commit:
more altar text formatting

Commit:
english altar text formatting complete

briaguya0 committed:
Merge pull request #276 from briaguya-ai/altar-text-formatting

Altar text formatting

Commit:
make gtg blocking guard check for card not bridge

briaguya0 committed:
Merge pull request #277 from briaguya-ai/gtg-guard

make gtg blocking guard check for card not bridge

PurpleHato committed:
FIX: Typo!

briaguya0 committed:
Merge pull request #278 from PurpleHato/RandoLocaFix

FIX: Typo!

PurpleHato committed:
FIX: Uppercases

briaguya0 committed:
Merge pull request #279 from PurpleHato/RandoLocaFix

FIX: Uppercases

PurpleHato committed:
FIX: Typo

PurpleHato committed:
TWEAK: Alter + some names

PurpleHato committed:
TWEAK: More caps!

PurpleHato committed:
ADD: Missing string

TWEAK more uppercases and namefixe
s

briaguya0 committed:
Merge pull request #280 from PurpleHato/RandoLocaFix

Even moooore caps

aMannus committed:
Hide nabooru death by covering her in flames

briaguya0 committed:
Merge pull request #281 from aMannus/spirit-temple-cs-skips-testing

Spirit Temple Nabooru Cutscene Skips

Commit:
Merge branch 'develop' into toirmd154

briaguya0 committed:
Merge pull request #282 from briaguya-ai/toirmd154

Toirmd154

Commit:
bandaid fix for death crash issue

briaguya0 committed:
Merge pull request #283 from briaguya-ai/death-crash-bandaid-fix

bandaid fix for death crash issue

aMannus committed:
Twinrova defeat cs skip

Skips the animation and manually calls the function to show the "beam" around the sisters

Commit:
fix crash

Commit:
fix caps to match

briaguya0 committed:
Merge pull request #286 from briaguya-ai/kak-impas-house-heart-piece

progressive stick capacity french capitalization fix

briaguya0 committed:
Merge pull request #285 from aMannus/rando-twinrova-defeat-cs-skip

Twinrova defeat CS skip

Commit:
fix great fairy reward mashing/shielding issue

briaguya0 committed:
Merge pull request #287 from briaguya-ai/thank-you-melon

fix great fairy reward mashing/shielding issue

PurpleHato committed:
TWEAK : Typo clé to Clé

PurpleHato committed:
TWEAK: Some Altar hints

TWEAK: Some capitals

PurpleHato committed:
TWEAK: Unmatching text + some cap again

briaguya0 committed:
Merge pull request #288 from PurpleHato/RandoLocaFix

Rando loca fix

PurpleHato committed:
TWEAK: More tweaks

briaguya0 committed:
Merge pull request #289 from PurpleHato/RandoLocaFix

TWEAK: More FR tweaks

Kenix3 reviewed (commented):
json.hpp in soh is redundant. We have a version in LUS for saves. We should probably use that.

Commit:
Merge branch 'develop' into toir-big-dev-merge

Commit:
fix build

briaguya0 committed:
Merge pull request #290 from briaguya-ai/toir-big-dev-merge

Toir big dev merge

Commit:
remove extra json.hpp, add hint

briaguya0 committed:
Merge pull request #291 from briaguya-ai/all-true-gamers

remove extra json.hpp, add hint

PurpleHato committed:
Update randomizer_item_tracker.cpp

PurpleHato committed:
TWEAK: Double Defense with RedParticles instead of white

briaguya0 committed:
Merge pull request #293 from PurpleHato/WarpSong

TWEAK: Double Defense with RedParticles instead of white

briaguya0 committed:
Merge pull request #292 from PurpleHato/ItemTracker

TWEAK: Increased Icon max size and spacing

Commit:
make sure we don't optimize out the check to ensure a spoilerfile exists

briaguya0 committed:
Merge pull request #294 from briaguya-ai/dont-optimize-out-file-check

make sure we don't optimize out the check to ensure a spoilerfile exists

Commit:
vanilla ganon boss key hint formatting

briaguya0 committed:
Merge pull request #295 from briaguya-ai/vanilla-ganon-boss-key-hint-formatting

vanilla ganon boss key hint formatting

PurpleHato committed:
TWEAK: FR- better way of the hero text

briaguya0 committed:
Merge pull request #297 from PurpleHato/RandoLocaFix

TWEAK: FR- better way of the hero text

Commit:
Merge branch 'develop' into toir-mergeweirdshot

Commit:
fix

Commit:
and again

briaguya0 committed:
Merge pull request #298 from briaguya-ai/toir-mergeweirdshot

Toir mergeweirdshot

leggettc18 committed:
Initializes dungeonsDone items in gSaveContext to 0.

leggettc18 committed:
Replaces sizeof calculation with a NUM_DUNGEONS constant.

briaguya0 committed:
Merge pull request #299 from leggettc18/rando_dungeonsdone_bug

Initializes dungeonsDone items in gSaveContext to 0.

leggettc18 committed:
Fixes Saria's Gift on the LW Bridge from getting killed when holding shield.

leggettc18 committed:
More airtight fix for Saria's Gift on the Bridge.

leggettc18 committed:
Lifts one of the conditions in the if statement a little higher to prevent unnecessary lookups of getItemId.

vaguerant committed:
Invalidate text box icon before drawing

Commit:
Merge branch 'develop' into toir-smalldevmerge

briaguya0 committed:
Merge pull request #302 from briaguya-ai/toir-smalldevmerge

Toir smalldevmerge

briaguya0 committed:
Merge pull request #303 from vaguerant/msgicon

Invalidate text box icon before drawing

leggettc18 committed:
Fixes the case where Saria's gift is an Ice Trap.

We still get the Ice Trap once, but never again. This does mean you can now hold R while walking in to avoid the ice trap, but everything else seems to work fine.

aMannus committed:
Initial commit

Might need changing when we change the settings in the future

leggettc18 committed:
Fixes Door of Time opening cutscene after warping with prelude.

briaguya0 committed:
Merge pull request #304 from aMannus/rando-settings-init-size-fix

Change rando settings size for first launch

briaguya0 committed:
Merge pull request #300 from leggettc18/rando_sariagiftcheck_bug

Fixes Saria's Gift on the LW Bridge from getting killed when holding shield

aMannus committed:
Initial waterfall skip

Very rudimentary way of doing things but it seems to work so :shrug:

briaguya0 committed:
Merge pull request #305 from leggettc18/rando_tot_song_fix

Fixes Door of Time opening cutscene after warping with prelude.

briaguya0 committed:
Merge pull request #306 from aMannus/rando-waterfall-cs-skip

Zora's River Waterfall Cutscene Skip

Baoulettes committed:
inital rework

Baoulettes committed:
fixed default rotation for 2D sprites

Baoulettes committed:
fix tab/space issues

briaguya0 committed:
Merge pull request #308 from Baoulettes/NewDropRuntimeFix

New drop runtime fix

Commit:
3d drops rando merge fix again

briaguya0 committed:
Merge pull request #309 from briaguya-ai/3d-drops-fix-too

3d drops rando merge fix again

leggettc18 committed:
Allows Impa to appear in the Lullaby check post drawbridge escape.

briaguya0 committed:
Merge pull request #310 from leggettc18/rando-impa-fix

Allows Impa to appear in the Lullaby check post drawbridge escape.

Commit:
Merge branch 'develop' into toir-dev-crash-fix-merge

briaguya0 committed:
Merge pull request #311 from briaguya-ai/toir-dev-crash-fix-merge

Toir dev crash fix merge

leggettc18 committed:
Changes Ganon's Trials Count setting to a checkbox

The checkbox is whether or not to skip all of them. Leaving the box unchecked will mean doing all of them. Eventually this will be switched back to a slider once we implement the logic for which trials start out completed.

leggettc18 committed:
Sets all Ganon's Trials to incomplete in new saves.

Fixes https://github.com/briaguya-ai/rando-issue-tracker/issues/131

Commit:
fix castle guards when oot throw cutscene has already played in rando

briaguya0 committed:
Merge pull request #312 from briaguya-ai/guards-post-ocarina-throw

fix castle guards when oot throw cutscene has already played in rando

Commit:
Merge branch 'develop' into another-day-another-dev-merge

briaguya0 committed:
Merge pull request #314 from briaguya-ai/another-day-another-dev-merge

Another day another dev merge

leggettc18 committed:
Properly removes the beams when trials are cleared.

leggettc18 committed:
Removes Question Mark from Skip Ganon's Trials UI.

leggettc18 committed:
Adds a todo comment about when to change back to slider.

briaguya0 committed:
Merge pull request #313 from leggettc18/rando-ganons-castle-fix

Ganon's Castle Fixes

Commit:
make deku seeds check for bullet bag

briaguya0 committed:
Merge pull request #315 from briaguya-ai/bullet-bag-check

make deku seeds check for bullet bag

PurpleHato committed:
Various tweaks

TWEAK: Altar Text
TWEAK: Hint names
TWEAK: Replace more problematic œ to oe

briaguya0 committed:
Merge pull request #316 from PurpleHato/RandoAltarHints

Various FRA tweaks

Commit:
upgrade ocarina on both child and adult equips

PurpleHato committed:
FIX: Jabu Item

Commit:
update equipped hookshot/longshot when obtained as other age

briaguya0 committed:
Merge pull request #317 from PurpleHato/RandoLocaFix

FIX: Jabu Item

briaguya0 committed:
Merge pull request #318 from briaguya-ai/items-of-the-ages

Items of the ages

Commit:
add hint

briaguya0 committed:
Merge pull request #319 from briaguya-ai/raveyard

in memoriam of disco dampe

Commit:
Merge branch 'develop' into devmergggg

briaguya0 committed:
Merge pull request #320 from briaguya-ai/devmergggg

Devmergggg

Commit:
don't give the bgs check without the claim check

briaguya0 committed:
Merge pull request #321 from briaguya-ai/no-bgs-check-without-claim-check

don't give the bgs check without the claim check

leggettc18 committed:
Skips Darunia Cutscene in Fire Temple

leggettc18 committed:
Added a TODO note about not skipping the cutscene.

There is a setting we will want to have eventually that will require this cutscene to not be skipped since it is used during a glitch.

briaguya0 committed:
Merge pull request #322 from leggettc18/rando-darunia-skip

Skips Darunia Cutscene in Fire Temple

Commit:
Merge branch 'develop' into mmmeerrrgggeeee

Commit:
remove todo

briaguya0 committed:
Merge pull request #323 from briaguya-ai/mmmeerrrgggeeee

Mmmeerrrgggeeee

Commit:
restore fast ocarina option in imgui that was lost in merge

briaguya0 committed:
Merge pull request #324 from briaguya-ai/bring-back-fast-ocarina-imgui

restore fast ocarina option in imgui that was lost in merge

GreatArgorath committed:
Fixes grey screen issue + tooltip for 2 handed shield

briaguya0 committed:
Merge pull request #325 from GreatArgorath/GreyFix

Fixes grey screen issue + tooltip for 2 handed shield

MelonSpeedruns reviewed (approved):
lgtm

PurpleHato reviewed (approved):
Legitimate! (trust me, I'm an engineer)

Commit:
Merge branch 'develop' into toir-hopefully-the-last-dev-merge

Commit:
update to use dg instead of g for textures in item tracker

PurpleHato committed:
TWEAK: Default color for cosmetic RAND button was not the corect one

Commit:
fix texture crash, remove unused item tracker code

briaguya0 committed:
Merge pull request #329 from briaguya-ai/toir-hopefully-the-last-dev-merge

Toir hopefully the last dev merge

Commit:
don't open mask shop until we get zelda's letter

briaguya0 committed:
Merge pull request #327 from PurpleHato/RANDIconFix

TWEAK: Default color for cosmetic RAND button was not the corect one

briaguya0 committed:
Merge pull request #330 from briaguya-ai/fix-mask-shop-logic

don't open mask shop until we get zelda's letter

briaguya0 committed:
Update README.md

leggettc18 committed:
Prevents "correct" chime under incorrect conditions.

leggettc18 committed:
Fixes typo in conditional and adds "bonk" sound effect.

"Bonk" sound is NA_SE_SY_OCARINA_ERROR and it plays when conditions for the Door of Time have not been met after playing Song of Time. This is only possible in rando's "Intended" Door of Time option, in which the Ocarina of Time and all 3 spritual stones are required to open the door, instead of the vanilla requirements of just having the song of time.

briaguya0 committed:
Merge pull request #331 from leggettc18/rando-dot-intended-fix

Fixes misleading sound effects with Door of Time "Intended" setting

Commit:
Merge branch 'develop' into toir-hopefullysmalldevmerge

briaguya0 committed:
Merge pull request #333 from briaguya-ai/toir-hopefullysmalldevmerge

Toir hopefullysmalldevmerge

Commit:
remove modify dpad equips toggle, replace with checks for dpad menu

Commit:
remove extra check

Commit:
add ability to hold c-up to assign to dpad when dpad menuing is enabled

Commit:
disable d-pad navigation on item menu when holding c-up to equip

Commit:
dpad+c-up stuff for equipment menu

briaguya0 committed:
Merge pull request #334 from briaguya-ai/better-dpad-items

Better dpad items

briaguya0 committed:
Merge branch 'HarbourMasters:develop' into testing-out-item-replacement

PurpleHato committed:
ADD: Checbox for songs colors

PurpleHato committed:
TWEAK: RandoColors for normal songs

Commit:
kind of quick and dirty but it works

PurpleHato committed:
Merge pull request #17 from briaguya-ai/make-checkbox-work

kind of quick and dirty but it works

PurpleHato committed:
TWEAK: Clarity of the tooltip

briaguya0 committed:
Merge pull request #336 from PurpleHato/RandoColor

ItemTracker: Vanilla + Rando colors for non warp songs

briaguya0 committed:
Merge branch 'HarbourMasters:develop' into testing-out-item-replacement

dcvz commented on soh/soh/OTRGlobals.cpp:
```diff
@@ -1423,3 +1445,138 @@ extern "C" void* getN64WeirdFrame(s32 i) {
     return &weirdFrameBytes[i + sizeof(n64WeirdFrames)];
 }
+
+extern "C" s16 GetItemModelFromId(s16 itemId) {
```
I think it might be useful to prefix randomizer specific methods. If they're more generally useful, they should be moved out of the Randomizer class. Tagged one but meann all new methods in this diff.

We have these examples:
```c++
OTRGetAspectRatio()
ResourceMgr_LoadCSByName
AudioPlayer_GetDesiredBuffered()

//
Randomizer_GetItemModelFromId()
Randomizer_LoadSettings()
```

dcvz commented on libultraship/libultraship/Lib/Fast3D/gfx_direct3d11.cpp:
```diff
@@ -413,7 +413,7 @@ static struct ShaderProgram *gfx_d3d11_create_and_load_new_shader(uint64_t shade
     size_t len, num_floats;
 
-    gfx_direct3d_common_build_shader(buf, len, num_floats, cc_features, false, d3d.current_filter_mode == THREE_POINT);
+    gfx_direct3d_common_build_shader(buf, len, num_floats, cc_features, false, d3d.current_filter_mode == FILTER_THREE_POINT);
```
not that it matters but was there a particular rando specific reason for renaming these?

briaguya0 replied:
not that i know of, i think that was a melon change

dcvz commented on libultraship/libultraship/Lib/Fast3D/gfx_dxgi.cpp:
```diff
@@ -265,6 +269,12 @@ static LRESULT CALLBACK gfx_dxgi_wnd_proc(HWND h_wnd, UINT message, WPARAM w_par
         case WM_KEYUP:
             onkeyup(w_param, l_param);
             break;
+        case WM_DROPFILES:
```
you only seem to use `fileName` here.. can it not be a local var?

dcvz commented on libultraship/libultraship/ImGuiImpl.cpp:
```diff
@@ -400,8 +400,14 @@ namespace SohImGui {
+        });
         Game::InitSettings();
+
+        CVar_SetS32("gRandoGenerating", 0);
```
once we get cvar setting out of LUS we can set these easily via  a`Randomizer::InitSettings()`

dcvz commented on libultraship/libultraship/Lib/Fast3D/gfx_sdl2.cpp:
```diff
@@ -106,7 +108,7 @@ static void set_fullscreen(bool on, bool call_callback) {
         window_width = mode.w;
         window_height = mode.h;
-        SDL_ShowCursor(false);
+        //SDL_ShowCursor(false);
```
Did we mean for this to stay in?

briaguya0 replied:
come to think of it, we probably meant for the one in dx to stay in too...

there hasn't been a real fix for https://github.com/HarbourMasters/Shipwright/issues/30 / https://github.com/HarbourMasters/Shipwright/issues/373, and it's something people notice quite often when using the randomizer menu in fullscreen

it'd be good to have a true solution worked out, but having the cursor hidden seems to cause more problems than not hiding it

dcvz replied:
Do we have a ticket for solving this?

dcvz commented on soh/include/z64item.h:
```diff
@@ -211,9 +211,9 @@ typedef enum {
-    /* 0x7B */ ITEM_INVALID_1,
-    /* 0x7C */ ITEM_INVALID_2,
-    /* 0x7D */ ITEM_INVALID_3,
+    /* 0x7B */ ITEM_SINGLE_MAGIC,
```
are these ported from zret?

briaguya0 replied:
they are not, we needed to add items for a few things that can be shuffled into the pool but don't have items in vanilla

dcvz commented on soh/include/z64save.h:
```diff
@@ -157,6 +173,17 @@ typedef struct {
     /* 0x1420 */ s16 worldMapArea;
     /* 0x1422 */ s16 sunsSongState; // controls the effects of suns song
     /* 0x1424 */ s16 healthAccumulator;
+    RandoSetting randoSettings[300];
```
again, might be best to prefix rando specific things

dcvz commented on soh/randomizer_generation.cpp:
```diff
@@ -0,0 +1,19 @@
+#include <Lib/spdlog/include/spdlog/spdlog.h>
+
+void GenerateRandomizer() {
```
any reason for this to not be static method on `Randomizer` class?

dcvz replied:
thoughts on this one @briaguya-ai

dcvz commented on soh/soh/OTRGlobals.cpp:
```diff
@@ -118,6 +122,8 @@ extern "C" void InitOTR() {
     DebugConsole_Init();
     Debug_Init();
+    Rando_Init();
+    InitItemTracker();
```
why not consolidate all into one `Rando_Init` method?

dcvz replied:
looks like github collapsing some comments made this one easy to miss @briaguya-ai

briaguya0 replied:
the item tracker and the main randomizer menu are separate imgui windows. i guess we could call over to the init from `randomizer_item_tracker` in `randomizer`, but i'm not sure if that is clearer. also, while the item tracker is currently in the randomizer menu, it works when playing vanilla as well.

i think it's worth considering where the item tracker should live instead of tying it in to randomizer even more.

dcvz replied:
Got it! Ok, currently its tied to the randomizer by name.. but makes sense to leave it separate in case we decide to move it out and then it should probably be renamed and moved away from the randomizer folder.

dcvz commented on soh/src/code/z_demo.c:
```diff
@@ -2073,29 +2090,43 @@ void Cutscene_HandleConditionalTriggers(GlobalContext* globalCtx) {
 
     if ((gSaveContext.gameMode == 0) && (gSaveContext.respawnFlag <= 0) && (gSaveContext.cutsceneIndex < 0xFFF0)) {
```
given that almost all branches in here only do something `if (!gSaveContext.n64ddFlag)` I think we can tidy up a bit

briaguya0 replied:
i agree, but i'm not quite sure what the best way to do it would be considering the ganon's tower part at the end

just moving the extra `if(!gSaveContext.n64ddFlag)` logic into the main conditionals would do a bit of cleanup work, but i feel like there might be a better way.

any suggestions?

dcvz replied:
It's tricky I see. let's at least move the `!gSaveContext.n64ddFlag` check to the outer ifs?
As far as I can tell from the logic that should not adversely affect things since they're all very explicit checks.

dcvz commented on soh/src/code/z_en_item00.c:
```diff
@@ -545,13 +548,17 @@ void func_8001DFC8(EnItem00* this, GlobalContext* globalCtx) {
 
     if (this->actor.params == ITEM00_HEART_PIECE) {
-        if (CVar_GetS32("gNewDrops", 0)) {
+        if ((CVar_GetS32("gNewDrops", 0) !=0) && !gSaveContext.n64ddFlag) {
```
don't think you need the `!=0`

briaguya0 replied:
yeah, definitely a merge thing there, the old new drops code had it, and i just added the 64ddflag check, good catch

dcvz commented on soh/src/code/z_horse.c:
```diff
@@ -72,7 +72,9 @@ void func_8006D0EC(GlobalContext* globalCtx, Player* player) {
         ASSERT(horseActor != NULL);
     } else if ((globalCtx->sceneNum == gSaveContext.horseData.scene) &&
-               (Flags_GetEventChkInf(0x18) != 0 || DREG(1) != 0)) {
+               (((Flags_GetEventChkInf(0x18) != 0) && (!gSaveContext.n64ddFlag ||
```
this is mostly an idea: so i see that sometimes several checks need to happen on rando.. would it make sense to extract checks into some centralized checks manager? so then in game code we can just do something like:
```c++
if (normalChecks && rando_canSpawnHorse())?
```

then 1. it's in a centralized place where people know to look for checks but also 2. you can give them a lot more descriptive names than they have now where you basically have to write lots of comments for it to be clear what you're checking and why -- while giving you a place to add those comments :)

will also make syncs with zret a bit easier ;)

briaguya0 replied:
definitely a good idea! would you think as a generalized pattern it makes sense to have `rando_canXYZ` checks include the rando check in them? so basically the first line of each one of those methods would be `if (!gSaveContext.n64ddFlag) return true;`

it also gets a little complicated based on where those methods would live and what information is available/what would need to be passed is as args, and there are definitely times where we want `normalChecks && randoChecks` and other times where we want `normalChecks || randoChecks`

i'm thinking this would be a good thing to make an issue for so we have time to think through all of the checks we're cleaning up

dcvz commented on soh/src/code/z_lights.c:
```diff
@@ -162,7 +162,9 @@ void Lights_BindAll(Lights* lights, LightNode* listHead, Vec3f* vec) {
     while (listHead != NULL) {
         info = listHead->info;
-        bindFuncs[info->type](lights, &info->params, vec);
+        if (info->type < 3) { // bandaid fix for death crash issue
```
is this a bug that was only fixed in this branch? can we get a ticket for this?

briaguya0 replied:
we've got <https://github.com/HarbourMasters/Shipwright/issues/654>

dcvz commented on soh/src/code/z_message_PAL.c:
```diff
@@ -1673,9 +1673,73 @@ void Message_OpenText(GlobalContext* globalCtx, u16 textId) {
+        // if we're rando'd and talking to a gossip stone
+        if (gSaveContext.n64ddFlag &&
+            textId == 0x2053 &&
+            GetRandoSettingValue(RSK_GOSSIP_STONE_HINTS) != 0 &&
```
for example here.. would be a very nice place to have used an extracted checker method.

briaguya0 replied:
made an issue for it and added it to the rando v2 milestone https://github.com/HarbourMasters/Shipwright/issues/741

dcvz reviewed (commented):
What a PR! 🤯 Great work on this @briaguya-ai, et al 💪 

I can't say I understood everything but I left some comments, some questions, some suggestions. I didn't really look in the 3drando folder itself. As I think that seems to come from another project.

Overall my intuition is that we can/should extract more things out of the main game code into Randomizer helper functions to 1. isolate randomizer specific logic and 2. give more context to the checks and 3. to not modify the game logic too much where avoidable (for zret syncing) but striking a good balance there is might be hard.

SoH has the concept of adding `filename_otr.cpp` files for custom SoH stuff.. something similar (`filename_rando.cpp`) might work for rando to hold additional methods that are needed like `GivePlayerRandoRewardMalon` in `z_en_ma1.c`.

briaguya0 commented:
> SoH has the concept of adding `filename_otr.cpp` files for custom SoH stuff.. something similar (`filename_rando.cpp`) might work for rando to hold additional methods that are needed like `GivePlayerRandoRewardMalon` in `z_en_ma1.c`.

definitely a good idea! made an issue for this one too https://github.com/HarbourMasters/Shipwright/issues/743

dcvz commented:
@briaguya-ai these two files currently live in the `soh` root, you probably want those elsewhere, right?

<img width="381" alt="Screenshot 2022-07-13 at 00 44 21" src="https://user-images.githubusercontent.com/2475932/178610843-4c657be8-c334-4139-81ce-7bd2472e7f43.png">
